src/httpserver.cpp

```diff
@@ -579,6 +566,9 @@ void HTTPRequest::WriteHeader(const std::string& hdr, const std::string& value)
 void HTTPRequest::WriteReply(int nStatus, const std::string& strReply)
 {
     assert(!replySent && req);
+    if (ShutdownRequested()) {
```

Comment:
To keep consistency and have lesser dependencies. Logically this should happen after http server interrupted.

Reply:
Ah I see, I'll do it.

Reply:
Actually I can't do that because that way the RPC `stop` reply won't get the `Connection: close` header — between `StartShutdown()` and `InterruptHTTPServer()` can go around 200ms:
https://github.com/bitcoin/bitcoin/blob/11e1ac3ae08535cefbd8235a8deb6cd100bcb2b1/src/bitcoind.cpp#L46-L54

Reply:
Ok. Let's use ShutdownRequested for now. Then we can switch to the flag if #13211 merged.

Reply:
I don't think that helps in this case.